src/Tuura/Plato/Translation.hs

```diff
@@ -27,41 +35,49 @@ instance Ord Signal
 
 -- TODO: Tidy up function, it looks ugly.
 addErrors :: (Eq a, Show a) => [ValidationError a] -> String
-addErrors errs =
-        if (unused errs) /= []
+addErrors errs = "Error\n" ++
+        (if unused /= []
         then "The following signals are not declared as input, output or internal: \n"
-             ++ unlines (map show (unused errs)) ++ "\n"
-        else "" ++
-        if (incons errs) /= []
+             ++ unlines (map show unused) ++ "\n"
+        else "") ++
+        (if incons /= []
         then "The following signals have inconsistent inital states: \n"
-             ++ unlines (map show (incons errs)) ++ "\n"
-        else "" ++
-        if (undefd errs) /= []
+             ++ unlines (map show incons) ++ "\n"
+        else "") ++
+        (if undefd /= []
         then "The following signals have undefined initial states: \n"
-             ++ unlines (map show (undefd errs)) ++ "\n"
-        else ""
+             ++ unlines (map show undefd) ++ "\n"
+        else "") ++
+        (if invVio /= []
+        then "The following state(s) are reachable but the invariant does not hold for them:\n"
+             ++ unlines (map show invVio) ++ "\n"
+        else "")
     where
-        unused es = [ a | UnusedSignal a <- es ]
-        incons es = [ a | InconsistentInitialState a <- es ]
-        undefd es = [ a | UndefinedInitialState a <- es ]
+        unused = [ a | UnusedSignal a <- errs ]
+        incons = [ a | InconsistentInitialState a <- errs ]
+        undefd = [ a | UndefinedInitialState a <- errs ]
+        invVio = [ a | InvariantViolated a <- errs ]
 
 validate :: Ord a => [a] -> CircuitConcept a -> ValidationResult a
-validate signs circuit
-    | unused ++ inconsistent ++ undef == [] = Valid
-    | otherwise = Invalid ((map UnusedSignal unused) ++ (map InconsistentInitialState inconsistent)
-                  ++ (map UndefinedInitialState undef))
+validate signs circuit = (validateInitialState signs circuit) `mappend` (validateInterface signs circuit)
```

Comment:
I wanted to use `mappend` so these are not confused with concepts, but `<>` would work and be neater so I will use it.